### PR TITLE
Modularization: Make ./etcd server a module

### PR DESCRIPTION
**Pull request by ptabor**
Modularization: Make ./etcd server specific module #12414

This PR finishes big reorg of code, putting all server related code in go.etcd.io/etcd/server/v3 module,
as agreed in: #12204 (comment).

After the change we have following structure:

 - server/  
   - auth/
   - embed/
   - etcdmain/
   - etcdserver/
   - lease/
   - mock/
   - mvcc/
   - proxy/
   - wal/
   - go.mod
   - go.sum
   - LICENSE
   - main.go
   - main_test.go

### CODE REVIEW

File: scripts/test_lib.sh [hunk 1]
```diff
@@ -105,14 +105,17 @@ function run_for_module {
...
-	go.etcd.io/etcd/raft/v3
-	go.etcd.io/etcd/client/v2
-	go.etcd.io/etcd/client/v3
-	go.etcd.io/etcd/v3
-	go.etcd.io/etcd/etcdctl/v3
-	go.etcd.io/etcd/tests/v3"
+  modules=(
+    go.etcd.io/etcd/api/v3
+    go.etcd.io/etcd/pkg/v3
+	  go.etcd.io/etcd/raft/v3
```
Comment (jingyih) [inline scripts/test_lib.sh:111]: The indentation here seems incorrect?

File: dummy.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main_test
+
+// MainTest package makes sure these packages stay as dependencies of the root
+// module (e.g. for sake of 'bom' generation).
+import (
+	_ "go.etcd.io/etcd/client/v2"            // keep
```
Comment (jingyih) [inline dummy.go:20]: Sorry I do not follow why we need to keep these two imports here?

File: scripts/test_lib.sh (hunk 1 above)
Comment (ptabor) [reply scripts/test_lib.sh]: Fixed. Tabs vs. spaces. Will need to add a checker to normalize bash scripts around 'spaces'.

File: dummy.go (hunk 2 above)
Comment (ptabor) [reply dummy.go:21]: 'go mod tidy' would remove them without these imports. Elaborated in the comment to the file. 

Bom is generated using: 
'/home/ptab/corp/etcd/tools/mod/.gobincache/github.com/coreos/license-bill-of-materials/@v/v0.0.0-20190913234955-13baff47494e/license-bill-of-materials' '--override-file' './bill-of-materials.override.json' 'go.etcd.io/etcd/api/v3/...' 'go.etcd.io/etcd/pkg/v3/...' 'go.etcd.io/etcd/raft/v3/...' 'go.etcd.io/etcd/client/v2/...' 'go.etcd.io/etcd/client/v3/...' 'go.etcd.io/etcd/server/v3/...' 'go.etcd.io/etcd/etcdctl/v3/...' 'go.etcd.io/etcd/tests/v3/...' 'go.etcd.io/etcd/v3

That needs to be called from a 'top' of the dependencies tree.

Comment (ptabor) [reply dummy.go:21]: Done fix: 
  https://github.com/etcd-io/etcd/pull/12421

**Review by jingyih — COMMENTED** [review]
lgtm. Thanks!

**Comment by tariq1890** [comment]
@jingyih We are looking forward to this change. When can we expect a release with this fix?

**Comment by jingyih** [comment]
> @jingyih We are looking forward to this change. When can we expect a release with this fix?

I am not aware of a release date for etcd v3.5 yet. There are some related information in the community meeting notes: https://docs.google.com/document/d/16XEGyPBisZvmmoIHSZzv__LoyOeluC5a4x353CX0SIM/edit

**Pull request merged**